src/server/pachyderm_test.go

```diff
@@ -2121,6 +2121,42 @@ func TestRecursiveCp(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestPipelineUniqueness(t *testing.T) {
```

Comment:
Not sure if you wanted to include this in this PR.  But either way the test looks good.